--- ezsolr.php.original	2014-09-24 13:06:52.000000000 +0200
+++ ezsolr.php	2014-09-24 12:51:32.000000000 +0200
@@ -643,6 +643,40 @@
 
             $docList[$languageCode] = $doc;
         }
+        
+        //@luca @patch
+        $generalPlugins = (array) $this->FindINI->variable( 'IndexPlugins', 'General' );
+        $classPlugins   = (array) $this->FindINI->variable( 'IndexPlugins', 'Class' );
+        if ( !empty( $generalPlugins ) )
+        {
+            foreach ( $generalPlugins as $pluginClassString )
+            {
+                if( !class_exists( $pluginClassString ) )
+                {
+                    eZDebug::writeError( "Unable to find the PHP class '$pluginClassString' defined for index time plugins for eZ Find", __METHOD__ );
+                    continue;
+                }
+                $plugin = new $pluginClassString;
+                if ( $plugin instanceof ezfIndexPlugin )
+                {
+                    $plugin->modify( $contentObject, $docList );
+                }
+            }
+        }
+
+        if (array_key_exists($contentObject->attribute( 'class_identifier' ), $classPlugins ) )
+        {
+            $pluginClassString = $classPlugins[$contentObject->attribute( 'class_identifier' )];
+            if ( class_exists( $pluginClassString ) )
+            {
+                $plugin = new $pluginClassString;
+                if ( $plugin instanceof ezfIndexPlugin )
+                {
+                        $plugin->modify( $contentObject, $docList );
+                }
+            }
+        }
+        //@luca @patch end
 
         $optimize = false;
         if ( $this->FindINI->variable( 'IndexOptions', 'DisableDirectCommits' ) === 'true' )
